RecoEgamma/ElectronIdentification/plugins/cuts/GsfEleEcalDrivenCut.cc

```diff
@@ -20,25 +15,61 @@ class GsfEleEcalDrivenCut : public CutApplicatorBase {
   }
 
 private:
-  const int ecalDrivenEB_, ecalDrivenEE_;// -1 ignore, 0 = fail ecalDriven, 1 =pass ecalDriven
+  static bool isValidCutVal(int val);
+
+private:
+  enum EcalDrivenCode{IGNORE=-1,FAIL=0,PASS=1};
+  const int ecalDrivenEB_, ecalDrivenEE_;
   const double barrelCutOff_;
+  
 };
 
 DEFINE_EDM_PLUGIN(CutApplicatorFactory,
 		  GsfEleEcalDrivenCut,
 		  "GsfEleEcalDrivenCut");
 
+
+ 
+GsfEleEcalDrivenCut::GsfEleEcalDrivenCut(const edm::ParameterSet& c) :
+  CutApplicatorBase(c),
+  ecalDrivenEB_(static_cast<EcalDrivenCode>(c.getParameter<int>("ecalDrivenEB"))),
+  ecalDrivenEE_(static_cast<EcalDrivenCode>(c.getParameter<int>("ecalDrivenEE"))),
+  barrelCutOff_(c.getParameter<double>("barrelCutOff"))
+{
+  if(!isValidCutVal(ecalDrivenEB_) || !isValidCutVal(ecalDrivenEE_)){
+    std::ostringstream errMsg;
+    errMsg<<"error in constructing GsfEleEcalDrivenCut"<<std::endl;
+    errMsg<<"values of ecalDrivenEB: "<<ecalDrivenEB_<<" and/or ecalDrivenEE: "<<ecalDrivenEE_<<" are invalid "<<std::endl;
+    errMsg<<"allowed values are IGNORE:"<<IGNORE<<" FAIL:"<<FAIL<<" PASS:"<<PASS;
+    throw edm::Exception(edm::errors::Configuration,errMsg.str());
+  }
+}
+
 CutApplicatorBase::result_type 
 GsfEleEcalDrivenCut::
 operator()(const reco::GsfElectronPtr& cand) const{ 
-  const bool ecalDriven =  std::abs(cand->superCluster()->position().eta()) < barrelCutOff_ ? 
+  const auto ecalDrivenRequirement =  std::abs(cand->superCluster()->position().eta()) < barrelCutOff_ ? 
     ecalDrivenEB_ : ecalDrivenEE_;
-  if(ecalDriven<0) return true;
-  else if(ecalDriven==0) return !cand->ecalDriven();
-  else return cand->ecalDriven();
+  if(ecalDrivenRequirement==IGNORE) return true;
+  else if(ecalDrivenRequirement==FAIL) return !cand->ecalDriven();
+  else if(ecalDrivenRequirement==PASS) return cand->ecalDriven();
+  else{  
+    std::ostringstream errMsg;
+    errMsg<<"error in "<<__FILE__<<" line "<<__LINE__<<std::endl;
+    errMsg<<"default option should not be reached, code has been updated without changing the logic, this needs to be fixed";
+    throw edm::Exception(edm::errors::LogicError,errMsg.str());
+  }
 }
 
 double GsfEleEcalDrivenCut::value(const reco::CandidatePtr& cand) const {
   reco::GsfElectronPtr ele(cand);
   return ele->ecalDriven();
 }
+
+bool GsfEleEcalDrivenCut::isValidCutVal(int val)
+{
+  if(val==IGNORE) return true;
```

Comment:
The compiler might optimize it out but otherwise they should make very different code (switch statement being much more efficient). I suppose it doesn't matter so much since it isn't called every event.

```c++
switch(val) {
case IGNORE:
case PASS:
case FAIL:
    return true;
    break;
default:
    break;
}
```